header/p2p/pb/extended_header_request.proto

```diff
@@ -3,7 +3,20 @@ syntax = "proto3";
 package p2p.pb;
 
 message ExtendedHeaderRequest {
-  uint64 origin = 1;
-  bytes hash = 2;
+  oneof data {
+    uint64 origin = 1;
+    bytes hash = 2;
+  }
   uint64 amount = 3;
 }
+
+enum StatusCode {
+  OK = 0;
+  NOT_FOUND = 1;
+  LIMIT_EXCEEDED = 2;
+};
```

Comment:
It is a good practice for proto error enum type is keeping zero value invalid. So if caller forget to set value and it becomes zero-value, it will not collide with actual status code. Example:
```
enum StatusCode {
    INVALID = 0;
    OK = 1;
    NOT_FOUND = 2;
    LIMIT_EXCEEDED = 3;
}
```

Reply:
Thank you. Done!